Create standalone `GutenbergKit` editor base class

oguzkocer opened the pull request:
## Summary
Isolates `GutenbergKit` editor from shared `EditorFragmentAbstract` to enable independent interface cleanup. Removes all `GutenbergKit` dependencies from `EditPostActivity`.

## Changes
- Create `GutenbergKitEditorFragmentBase` as standalone copy of `EditorFragmentAbstract`
- Switch `GutenbergKitEditorFragment` to extend standalone base class
- Remove all `GutenbergKit` references from `EditPostActivity` (220+ lines removed)
- Clean separation: `EditPostActivity` handles Aztec + Gutenberg, `GutenbergKitActivity` handles `GutenbergKit`

## Impact
- **Zero functional changes** - pure architectural isolation
- **Enables aggressive cleanup** of unused `EditorFragmentListener` methods in follow-up PRs
- **Clean architecture** with no coupling between editor types

oguzkocer committed:
Copy EditorFragmentAbstract as GutenbergKitEditorFragmentBase

Creates a GutenbergKit-specific copy of EditorFragmentAbstract to enable
independent cleanup without affecting shared editor infrastructure.

This is a direct copy with no changes - subsequent commits will adapt it
for GutenbergKit-specific needs and remove unused functionality.

oguzkocer committed:
Switch GutenbergKit to use standalone editor base class

Creates GutenbergKit-specific editor architecture by switching from
EditorFragmentAbstract to GutenbergKitEditorFragmentBase. This enables
independent cleanup of GutenbergKit editor interfaces without affecting
shared editor infrastructure.

Changes:
- Update GutenbergKitEditorFragment to extend GutenbergKitEditorFragmentBase
- Update GutenbergKitActivity to use GutenbergKit-specific interfaces
- Adapt package structure and imports for standalone architecture
- Retain original TrackableEvent enum and EditorFragmentNotAddedException
- Update exception handling to use original shared exception types
- Remove unnecessary interface and exception duplication

Impact: Zero functional changes - creates foundation for aggressive cleanup

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oguzkocer committed:
Adds GutenbergKitEditorFragmentBase.GutenbergKitEditorFragmentBase

oguzkocer committed:
Remove GutenbergKit dependencies from EditPostActivity

Clean separation after standalone base class creation - EditPostActivity
no longer handles GutenbergKit fragments, which are now managed by
GutenbergKitActivity independently.

Changes:
- Remove GutenbergKit fragment creation and initialization logic
- Remove GutenbergKit-specific feature flags and imports
- Remove unused isFinishing parameters throughout call chain
- Simplify editor type detection and menu handling
- Remove GutenbergKit settings creation methods

Impact: ~150 lines removed, cleaner architecture with zero coupling
between EditPostActivity and GutenbergKit components.

dcalhoun commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.kt:
```diff
@@ -1769,10 +1740,6 @@ class EditPostActivity : BaseAppCompatActivity(), EditorFragmentActivity, Editor
                 // toggle HTML mode
                 if (editorFragment is AztecEditorFragment) {
                     (editorFragment as AztecEditorFragment).onToolbarHtmlButtonClicked()
-                } else if (editorFragment is GutenbergEditorFragment) {
-                    (editorFragment as GutenbergEditorFragment).onToggleHtmlMode()
```
This appears to be erroneously removed. It relates to Gutenberg Mobile, not GutenbergKit. In my testing, tapping _More_ (three vertical dots icon in the top-right corner) → _HTML Mode_ does not toggle the editor from visual to code editing mode.

oguzkocer replied:
Sorry about that! Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/22149.

dcalhoun reviewed (changes requested):
The changes look good. Thank you for improving this code. 

I tested the three editors—GutenbergKit, Gutenberg Mobile, and Aztec—using the build from https://github.com/wordpress-mobile/WordPress-Android/pull/22144. I primarily focused on text editing, media uploads, and navigating menu items in the more menu. 

As noted in an inline comment, I noticed the _HTML Mode_ toggle in Gutenberg Mobile no longer functions. Other than that, I did not observe any regressions from these changes.

dcalhoun reviewed (approved):
After speaking with @oguzkocer in chat, we'll address https://github.com/wordpress-mobile/WordPress-Android/pull/22143#discussion_r2286075364 in a follow PR. Approving these changes so that we can merge this and related PRs with minimal rebasing required.

oguzkocer commented:
This PR is part of a rebase train. The build has been verified by CI before the last rebase, and to save on time, we are going to land all these PRs at once using admin-merge.

Note that any issues in the PR will be addressed in a follow up PR. I'll reply to each comment with the hash that addresses it.
